bip-0352.mediawiki

```diff
@@ -0,0 +1,475 @@
+
+The one exception is script path spends that use NUMS point ''H'' as their internal key (where ''H = lift_x(0x50929b74c1a04954b78b4b6035e97a5e078a5a0f28ec96d547bfee9ace803ac0)'' which is constructed by taking the hash of the standard uncompressed encoding of the secp256k1 base point ''G'' as X coordinate, see [https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#constructing-and-spending-taproot-outputs BIP341: Constructing and spending Taproot outputs] for more details), in which case the output will be skipped for the purposes of shared secret derivation<ref name="why_ignore_h">'''Why skip outputs with H as the internal taproot key?''' If use cases get popularized where the taproot key path cannot be used, these outputs can still be included without getting in the way of making a silent payment, provided they specifically use H as their internal taproot key.</ref>.
+
+''' P2WPKH '''
+
+The sender performs the tweak using the private key for the output and the receiver obtains the public key from the ''witness''.
+
+''' P2SH-P2WPKH '''
+
+The sender performs the tweak using the private key for the nested ''P2WPKH'' output and the receiver obtains the public key from the ''witness''.
+
+''' P2PKH '''
+
+The sender performs the tweak using the private key for the output and SHOULD sign using the standard script template:
+
+
+    scriptSig: <Signature> <Public Key>
+
+
+The receiver obtains the public key from the ''scriptSig''. The receiver MUST parse the ''scriptSig'' for the public key, even if the ''scriptSig'' is non-standard (e.g. <code><dummy> OP_DROP <Signature> <Public Key></code>). This is to address the [https://en.bitcoin.it/wiki/Transaction_malleability third-party malleability of ''P2PKH'' ''scriptSigs''].
+
+''' P2PK '''
+
+The sender performs the tweak using the private key for the output and the receiver obtains the public key from the ''scriptPubKey''.
+
+=== Outpoint hash ===
+
+The sender and receiver MUST calculate an outpoints hash for the transaction in the following manner:
+
+* Let ''A = A<sub>0</sub> + A<sub>1</sub> + ... + A<sub>n</sub>'', where each ''A<sub>i</sub>'' is the public key of an input from the ''[[#inputs-for-shared-secret-derivation|Inputs For Shared Secret Derivation]]'' list
+* Let ''outpoint = outpoint<sub>0</sub>'',  where ''outpoint<sub>0</sub>'' is the lowest outpoint lexicographically by txid and vout used in the transaction<ref name="why_lowest_outpoint"></ref>
+* Let ''outpoint_hash = hash(outpoint || A)''
```

Comment:
There's really no need to have _outpoint_ as a separate variable anymore. Also, _outpoint<sub>0</sub>_ might be confused with the first outpoint, how about _outpoint<sub>L</sub>_?
```suggestion
* Let ''outpoint_hash = hash(outpoint<sub>L</sub> || A)'', where ''outpoint<sub>L</sub>'' is the lowest outpoint lexicographically by txid and vout used in the transaction<ref name="why_lowest_outpoint"></ref>
```

Reply:
Thanks for the suggestion, this reads better